ocaml/xapi/agility.ml

```diff
@@ -72,6 +72,13 @@ let caching_vm_t_assert_agile ~__context (ok_srs, ok_networks) vm vm_t =
   if vm_t.API.vM_VUSBs <> [] then
     raise (Api_errors.Server_error
              (Api_errors.vm_has_vusbs, [Ref.string_of vm]));
+
+  List.iter (fun vif ->
+      if Db.VIF.get_reserved_pci ~__context ~self:vif <> Ref.null then
```

Comment:
This would only work if the VM is running, but we also want to check agility for halted VMs. You could use `Xapi_network_sriov_helpers.is_sriov_network` on the VIF's network to check if it is SR-IOV. May be best to do this in `check_vif` below, since we already get the network there, and take advantage of caching.

Reply:
Ah, yes, I forget `VIF.reserved_pci` will be clear when shutting down.

Reply:
It's still there...